### PR TITLE
MultitaskReadout: only iterate over tasks in batch

### PR DESCRIPTION
**Pull request by nandahkrishna**
The current `MultitaskReadout.forward` and `forward_varlen` loop over every entry in `self.readout_specs`, testing each with `torch.any(mask)` to skip absent modalities. This scales the forward cost with the total size of the readout registry even though each batch only touches at most batch_size * n_out distinct modalities.                                
                                                                         
At large modality counts (e.g. per-session readouts over thousands of sessions, or large brainset pools), the empty-mask `==` + `torch.any`operations slow down the forward pass quite a bit.

This PR makes `MultitaskReadout` iterate over modalities or tasks present within the batch, skipping IDs that are for whatever reason non-existent in readout_specs for the MultitaskReadout (e.g., padding or deliberately removed tasks that still exist in the collated batch -- just for safety). The mapping between readout IDs and names is cached upon initial creation, along with storing the projection matrices in the dictionary.

Benchmarking:
| `readout_specs` | no-fix fwd | patched fwd | **fwd speedup** | no-fix total | patched total | **total speedup** |
|---:|---:|---:|---:|---:|---:|---:|
|   ~21 | 15.0 |   —   |   —   |  49.9 |   —   |   —   |
|   ~30 | 16.1 | 15.6 | 1.03× |  53.6 |  53.1 | 1.01× |
|  ~120 | 28.0 | 25.5 | 1.10× |  85.5 |  82.9 | 1.03× |
| ~1020 | 53.0 | 31.6 | **1.68×** | 122.8 | 101.7 | **1.21×** |

At 10k datasets/readouts, I expect this to yield even up to 4x speedups.

Hoping to land this soon as we need it for a project, would appreciate reviews!

## Summary by CodeRabbit

* **Refactor**
  * Optimized multitask readout layer to process only active tasks present in the current batch, reducing unnecessary computation and improving performance for models with variable task configurations.

### CODE REVIEW

**Review by milosobral — APPROVED** [review]
Looks good!

File: torch_brain/nn/multitask_readout.py [hunk 1]
```diff
@@ -31,8 +31,10 @@ def __init__(
 
         # Create a bunch of projection layers. One for each readout
         self.projections = nn.ModuleDict({})
-        for readout_id, spec in self.readout_specs.items():
-            self.projections[readout_id] = nn.Linear(dim, spec.dim)
+        self._readout_id_to_name = {}
+        for readout_name, readout_spec in self.readout_specs.items():
+            self.projections[readout_name] = nn.Linear(dim, readout_spec.dim)
+            self._readout_id_to_name[readout_spec.id] = readout_name
```
[the remark this answers is not in the document]
Comment (nandahkrishna) [reply torch_brain/nn/multitask_readout.py:37]: This should be done in modality registry if at all, not here I'd say.

**Comment by nandahkrishna** [comment]
Thanks for the quick review -- merging now but if there are any issues I guess we can fix forward!
